src/js/utils/string-utils.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * Useful string handling utilities.
+ */
+define(['underscore'], function(_) {
+    'use strict';
+
+    var interpolateEscape, escapeInterpolateOptions, HTML;
+
+    /**
+     * Creates an HTML snippet.
+     *
+     * @param {string} text The HTML for the text.
+     * @constructor
+     */
+    function HTMLSnippet(text) {
+        this.text = text;
+    }
+    HTMLSnippet.prototype.toString = function() {
+        return this.text;
+    };
+
+    HTML = function(text) {
+        return new HTMLSnippet(text);
+    };
+
+    /**
+     * Returns an HTML-escaped text string by substituting values into
+     * a format string. Parameters are specified as tokens wrapped by
+     * curly braces, e.g. 'Hello {name}'. The tokens are replaced
+     * by the attribute of the same name from the specified options.
+     *
+     * Note: values will be rendered using their toString methods and then
+     * HTML-escaped. The only exception is that instances of the class HTML
+     * are rendered without escaping as their contract declares that they are
+     * already valid HTML.
+     *
+     * Example:
+     *   StringUtils.interpolateEscape(
+     *       'You are enrolling in {spanStart}{courseName}{spanEnd}',
```

Comment:
As we discussed, gettext is a Django i18n function, so it is a little strange to embed that into a general purpose JavaScript library that might be used in IDAs that are not Django based. I'm going to leave the example as is, but then add a description about the needs of i18n.